torchvision/ops/stochastic_depth.py

```diff
@@ -0,0 +1,56 @@
+import torch
+from torch import nn, Tensor
+
+
+def stochastic_depth(input: Tensor, mode: str, p: float, training: bool = True) -> Tensor:
+    """
+    Implements the Stochastic Depth from `"Deep Networks with Stochastic Depth"
+    <https://arxiv.org/abs/1603.09382>`_ used for randomly dropping residual
+    branches of residual architectures.
+
+    Args:
+        input (Tensor[N, ...]): The input tensor or arbitrary dimensions with the first one
+                    being its batch i.e. a batch with ``N`` rows.
+        mode (str): ``"batch"`` or ``"row"``.
+                    ``"batch"`` randomly zeroes the entire input, ``"row"`` zeroes
+                    randomly selected rows from the batch.
+        p (float): probability of the input to be zeroed.
+        training: apply dropout if is ``True``. Default: ``True``
+
+    Returns:
+        Tensor[N, ...]: The randomly zeroed tensor.
+    """
+    if p < 0.0 or p > 1.0:
+        raise ValueError("drop probability has to be between 0 and 1, but got {}".format(p))
+    if not training or p == 0.0:
+        return input
+
+    survival_rate = 1.0 - p
+    if mode == "batch":
+        keep = torch.rand(size=(1, ), dtype=input.dtype, device=input.device) < survival_rate
+    elif mode == "row":
+        keep = torch.rand(size=(input.size(0),), dtype=input.dtype, device=input.device) < survival_rate
+        keep = keep[(None, ) * (input.ndim - 1)].T
+    else:
+        raise ValueError("mode has to be either 'batch' or 'row', but got {}".format(mode))
+    return input / survival_rate * keep
```

Comment:
nit: here are some potential micro-optimizations, as we don't need to divide the whole input by `survival_rate`, but we can do it only in the `keep`. This is similar to what is done in the [PyTorch implementation of dropout](https://github.com/pytorch/pytorch/blob/913c1f83f49f9e1e2a494186cc0069d780cee852/aten/src/ATen/native/Dropout.cpp#L62)

```suggestion
    if mode not in ["batch", "row"]:
        raise ValueError("mode has to be either 'batch' or 'row', but got {}".format(mode))
    size = [1] * input.ndim
    if mode == "row":
       size[0] = input.shape[0]
   noise = torch.empty(size, dtype=input.dtype, device=input.device)
   noise = noise.bernoulli_(survival_rate).div_(survival_rate)
   return input * noise
```

What I like about this implementation is that it makes it even clearer what the differences are between the different modes. But it's a minor difference in clarity, so I'm not set to stone on my proposal.